mat64/dense_arithmetic.go

```diff
@@ -351,6 +351,7 @@ func (m *Dense) Dot(b Matrix) float64 {
 }
 
 // Mul takes the matrix product of a and b, placing the result in the receiver.
+// If a or b are also the receiver, a workspace will be allocated.
```

Comment:
I don't think any mention of storing the result should be made, that confuses the matter, I will s/workspace/temporary workspace memory/. The text is purely there to prevent people from doing silly things like a.Mul(a, a) in a tight loop and wondering why things are slow.

Here might actually be a good place for pools though. I can dig that code out and make it work again.